src/lib/mina_base/parties.ml

```diff
@@ -908,11 +915,44 @@ let weight (parties : t) : int =
     ]
 
 module Valid = struct
-  module Stable = Stable
+  module S = Stable
+
+  module Verification_key_hash = struct
+    [%%versioned
+    module Stable = struct
+      module V1 = struct
+        type t = Zkapp_basic.F.Stable.V1.t
+        [@@deriving sexp, compare, equal, hash, yojson]
+
+        let to_latest = Fn.id
+      end
+    end]
+
+    [%%define_locally Stable.Latest.(equal)]
+  end
 
-  type t = Stable.Latest.t
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      type t =
+        { parties : S.V1.t
```

Comment:
Can you make this type private to external users? While it's public, it's hard to assert any invariants on it.